src/Spago/Build.hs

```diff
@@ -83,10 +83,22 @@ build BuildOptions{..} maybePostBuild = do
     NoInstall -> pure ()
   let allGlobs = Packages.getGlobs deps depsOnly configSourcePaths <> sourcePaths
       buildAction = do
-        Purs.compile allGlobs pursArgs
-        case maybePostBuild of
-          Just action -> action
-          Nothing     -> pure ()
+        case alternateBackend of 
+          Nothing ->
+              Purs.compile allGlobs pursArgs
+          Just backend -> do
+              Purs.compile allGlobs $
+                pursArgs ++
```

Comment:
`purs` will just panic if you pass it the same argument twice, so here we should check if the user is already passing `--codegen`, and then do something about it (options: either override what the user is doing or just fail and explain the situation)

Reply:
I think failing is preferable, that way the user knows, that he can save himself the effort of passing the flag